llpc/tool/amdllpc.cpp

```diff
@@ -373,139 +373,72 @@ static Result init(int argc, char *argv[], ICompiler **ppCompiler) {
-    // which arguments are not option names.
-    static const char *DefaultOptions[] = {
-        // Name                                Option
-        "-gfxip",
-        "-gfxip=8.0.0",
-        "-unroll-max-percent-threshold-boost",
-        "-unroll-max-percent-threshold-boost=1000",
-        "-pragma-unroll-threshold",
-        "-pragma-unroll-threshold=1000",
-        "-unroll-allow-partial",
-        "-unroll-allow-partial",
-        "-simplifycfg-sink-common",
-        "-simplifycfg-sink-common=false",
-        "-amdgpu-vgpr-index-mode",
-        "-amdgpu-vgpr-index-mode", // Force VGPR indexing on GFX8
-        "-amdgpu-atomic-optimizations",
-        "-amdgpu-atomic-optimizations", // Enable atomic optimizations
-        "-use-gpu-divergence-analysis",
-        "-use-gpu-divergence-analysis", // Use new divergence analysis
-        "-filetype",
-        "-filetype=obj", // Target = obj, ELF binary; target = asm, ISA assembly text
-    };
-
-    // Build new arguments, starting with those supplied in command line
-    std::vector<const char *> newArgs;
-    for (int i = 0; i < argc; ++i)
-      newArgs.push_back(argv[i]);
-
-    static const size_t DefaultOptionCount = sizeof(DefaultOptions) / (2 * sizeof(DefaultOptions[0]));
-    for (unsigned optionIdx = 0; optionIdx != DefaultOptionCount; ++optionIdx) {
-      const char *name = DefaultOptions[2 * optionIdx];
-      const char *option = DefaultOptions[2 * optionIdx + 1];
-      size_t nameLen = strlen(name);
-      bool found = false;
-      const char *arg = nullptr;
-      for (int i = 1; i < argc; ++i) {
-        arg = argv[i];
-        if (strncmp(arg, name, nameLen) == 0 &&
-            (arg[nameLen] == '\0' || arg[nameLen] == '=' || isdigit((int)arg[nameLen]))) {
-          found = true;
-          break;
-        }
-      }
-
-      if (!found)
-        newArgs.push_back(option);
-      else if (optionIdx == 0) // Find option -gfxip
-      {
-        size_t argLen = strlen(arg);
-        if (argLen > nameLen && arg[nameLen] == '=') {
-          // Extract tokens of graphics IP version info (delimiter is ".")
-          const unsigned len = argLen - nameLen - 1;
-          char *gfxIp = new char[len + 1];
-          memcpy(gfxIp, &arg[nameLen + 1], len);
-          gfxIp[len] = '\0';
-
-          char *tokens[3] = {}; // Format: major.minor.step
-          char *token = std::strtok(gfxIp, ".");
-          for (unsigned i = 0; i < 3 && token; ++i) {
-            tokens[i] = token;
-            token = std::strtok(nullptr, ".");
-          }
-
-          ParsedGfxIp.major = tokens[0] ? std::strtoul(tokens[0], nullptr, 10) : 0;
-          ParsedGfxIp.minor = tokens[1] ? std::strtoul(tokens[1], nullptr, 10) : 0;
-          ParsedGfxIp.stepping = tokens[2] ? std::strtoul(tokens[2], nullptr, 10) : 0;
-
-          delete[] gfxIp;
+    // Before we get to LLVM command-line option parsing, we need to find the -gfxip option value.
+    for (int i = 1; i != argc; ++i) {
```

Comment:
int -> unsigned, better to use unsigned unless we want to a signed comparison.

Reply:
I tried that originally, but that gives a warning on one compiler (an error because we have warnings-as-errors) because it is compared with argc, which is an int.

Reply:
I see. I didn't notice argc was a signed one.